test/test/test_network_manipulation.rb

```diff
@@ -376,7 +387,7 @@ module Test
                 end
 
                 it "fails if the plan can't be deployed" do
-                    plan.add_mission_task(@cmp_m.as_plan)
+                    plan.add_mission_task(@cmp_m.use("srv" => @task_m).as_plan)
```

Comment:
I dont remember adding this line...Anyways, this is failing, somehow instead of UnexpectedErrors I am getting the actual error TaskAllocationFailed. Same thing if I remove this specific change. Still looking into it....

Reply:
This was an issue with the return value of apply_requirements_modifications#syskit_apply_async_resolution_results. I had to update it to be more informative instead of reporting either nil or error. Doing a full pass in the unit tests right now, then I will push this change